component/LinkableText/LinkableText.tsx

```diff
@@ -2,11 +2,11 @@ import Link from "next/link";
 
 const LinkableText = ({text}: {text: string}) => {
     const sanitizeNewLine = text?.split('\n')
-    const regexUrl = /^((?:https?:\/\/)?[^./]+(?:\.[^./]+)+(?:\/.*)?)$/
+    const regexUrl = /(https?\:\/\/)?([^\.\s]+)?[^\.\s]+\.[^\s]+/gi
 
     const textComponent = sanitizeNewLine?.map((textline, index) => {
         const textLineSplit = textline?.split(' ')
-        return <p key={textline}>
+        return <p className="font-normal text-sm" key={textline}>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

It's worth noting that the `split(' ')` method used here to split the text into words may not handle all cases correctly. For instance, it won't correctly handle multiple spaces between words or tabs. Consider using a more robust method for splitting the text into words, such as `split(/\s+/)`.
```diff
- const textLineSplit = textline?.split(' ')
+ const textLineSplit = textline?.split(/\s+/)
```

Also, the `key` prop in React should be unique among siblings. Using `textline` as a key could potentially lead to issues if there are duplicate lines in the text. Consider using the `index` from the map function as a key instead.
```diff
- return <p className="font-normal text-sm" key={textline}>
+ return <p className="font-normal text-sm" key={index}>
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->